app/src/main/java/com/kickstarter/ui/data/CommentCardData.kt

```diff
@@ -28,8 +28,31 @@ class CommentCardData(
     }
 
     companion object {
-        fun builder() = CommentCardData.Builder()
+        fun builder() = Builder()
     }
 
-    fun toBuilder() = CommentCardData.Builder(this.comment, this.commentCardState, this.commentableId, this.project)
+    fun toBuilder() = Builder(this.comment, this.commentCardState, this.commentableId, this.project)
+
+    override fun equals(other: Any?): Boolean {
+        var equals = super.equals(other)
+
+        if (other is CommentCardData) {
+            val obj: CommentCardData = other
+            equals = (
+                obj.comment?.equals(this.comment) == true &&
+                    obj.commentCardState == this.commentCardState &&
+                    obj.project?.equals(this.project) == true
+                )
+        }
+
+        return equals
+    }
+
+    override fun hashCode(): Int {
```

Comment:
Why are we overriding the `hasCode()` do we need it? 
also the value "31" there what does it mean?

Reply:
That was generated by my IDE. I don't think it's used, will remove it